### PR TITLE
implement chat messages

### PR DESCRIPTION
**Pull request by aviau**

### CODE REVIEW

File: src/icq/client/events.rs [hunk 1]
```diff
@@ -241,5 +248,5 @@ pub struct HistDlgStateMessageChat {
     // }
     pub sender: String, // The sender's sn
     pub name: String,   // The chat name
-    pub live: bool,
+                        //pub live: Option<bool>,
```
Comment (leameow) [inline src/icq/client/events.rs]: why the indent?

File: src/icq/poller.rs [hunk 2]
```diff
@@ -139,23 +143,49 @@ pub async fn process_event_hist_dlg_state(
...
+        let chat_input = ChatInput {
+            chat_sn: chat_sn.clone(),
+            author_sn,
+            author_friendly,
+            text: message.text.clone(),
+            time: message.time,
+        };
+
+        tx.handle_proxy(&account_info.handle)
+            .exec(move |_plugin, protocol_data| {
```
Comment (leameow) [inline src/icq/poller.rs]: nit, if you only use connection then `connection_proxy` can be used.

File: src/lib.rs [hunk 3]
```diff
@@ -42,6 +42,15 @@ pub mod chat_states {
     pub const JOINED: &str = "joined";
 }
 
+#[derive(Debug, Clone)]
+pub struct ChatInput {
```
Comment (leameow) [inline src/lib.rs]: Since `chat` in purple means something specific, how about `MsgInfo` or something alike?

Comment (aviau) [reply src/lib.rs]: done

File: src/icq/poller.rs (hunk 2 above)
Comment (aviau) [reply src/icq/poller.rs]: ah yes. Done

File: src/icq/client/events.rs (hunk 1 above)
Comment (aviau) [reply src/icq/client/events.rs]: rustfmt :shrug: 

I worked around it.

**Comment by leameow** [comment]
lgtm!

**Pull request merged**